lua/nvim-dap-virtual-text/virtual_text.lua

```diff
@@ -109,7 +111,9 @@ function M.set_virtual_text(stackframe, options)
         end
 
         if in_scope then
-          variables[name] = nil
+          if options.only_first_definition and not options.all_references then
+            variables[name] = nil
```

Comment:
This was actually a cheat since I'm not really doing proper shadowing analysis at the moment. ALternative could be #24